flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecLookupJoin.java

```diff
@@ -211,6 +214,64 @@ public Transformation<RowData> translateToPlanInternal(
                 planner, config, upsertMaterialize, lookupKeyContainsPrimaryKey);
     }
 
+    @Override
+    protected Transformation<RowData> createKeyOrderedAsyncLookupJoin(
+            Transformation<RowData> inputTransformation,
+            RelOptTable temporalTable,
+            ExecNodeConfig config,
+            ClassLoader classLoader,
+            Map<Integer, LookupJoinUtil.LookupKey> allLookupKeys,
+            AsyncTableFunction<Object> asyncLookupFunction,
+            RelBuilder relBuilder,
+            RowType inputRowType,
+            RowType tableSourceRowType,
+            RowType resultRowType,
+            boolean isLeftOuterJoin,
+            LookupJoinUtil.AsyncLookupOptions asyncLookupOptions) {
+        int[] shuffleKeys = inputUpsertKey;
+        // normally upsertKeys could not be null. If the job is restored from exec plan then
+        // upsertKeys could be null
+        if (shuffleKeys == null || shuffleKeys.length == 0) {
+            shuffleKeys = IntStream.range(0, inputRowType.getFieldCount()).toArray();
+        }
+
+        RowDataKeySelector keySelector =
+                getKeySelector(false, shuffleKeys, classLoader, inputRowType);
+
+        Transformation<RowData> partitionedTransform =
+                createPartitionTransformation(keySelector, inputTransformation, false, config);
+
+        StreamOperatorFactory<RowData> operatorFactory;
+
+        operatorFactory =
+                createAsyncLookupJoin(
```

Comment:
The operator impl is not ready, right? Can we throw an exception here instead of using a wrong operator?